pkg/controller/controller_pod.go

```diff
@@ -108,6 +108,11 @@ func (tc *TFJobController) reconcilePods(
 				})
 			}
 
+			// Set restart policy
+			if spec.RestartPolicy != "" && spec.RestartPolicy != "ExitCode" {
```

Comment:
As we discussed, k8s will set RestartPolicy for us then I think there is no need to check if the restartpolicy is empty

Reply:
SGTM.